music21/humdrum/spineParser.py

```diff
@@ -2119,7 +2122,27 @@ def getAllOccurring(self):
         return retEvents
 
 
-def hdStringToNote(contents):
+@lru_cache(maxsize=256)
+def _noteMemos(contents):
```

Comment:
new music21 routines require typing.  Can you type the input and output of this routine?

Reply:
Give some docs about what this does.